src/components/shared/Map/MapPointer.tsx

```diff
@@ -10,17 +10,48 @@ interface MapPointerProps {
   center: [number, number];
   radius: number;
   text: string;
-  color: string;
   sensorData: SensorData;
 }
 
-function MapPointer({
-  center,
-  radius,
-  text,
-  color,
-  sensorData
-}: MapPointerProps) {
+const colors = ['#6ebc2a', '#c4d83f', '#ffe500', '#f1aa41', '#b91414'];
+
+type PollutionLevels = Array<{
+  name: string;
+  values: number[];
+}>;
+
+const pollutionLevels: PollutionLevels = [
+  {
+    name: 'PM10',
+    values: [0, 40, 80, 120, 300]
+  },
+  {
+    name: 'PM2.5',
+    values: [0, 25, 50, 100, 300]
+  }
+];
+
+function colorCircle(sensorData: SensorData, pollutionName: string): string {
+  let color: string = '#909090';
+  if (pollutionName != null) {
+    const pollutionValue = sensorData.data.find(
+      item => item.name === pollutionName
+    );
+    const scale = pollutionLevels.find(item => item.name === pollutionName);
+    if (pollutionValue != null && scale != null) {
+      for (let i = 0; i < scale.values.length; i += 1) {
+        const treshold = scale.values[i];
+        if (treshold != null && pollutionValue.value < treshold) {
+          break;
+        }
+        color = colors[i] ?? '#909090';
+      }
+    }
+  }
+  return color;
```

Comment:
```suggestion
    const idx = pollutionLevels[pollutionName].find(threshold => pollutionValue.value < threshold);
    return COLORS.pollutionScale[idx];
```
You will be able to simplify code as shown in this suggestion after applying changes described in the previous comments.